docs/madr/decisions/070-resource-identifier.md

```diff
@@ -0,0 +1,332 @@
+
+Envoy admin pages use `:`
+```bash
+$ curl http://localhost:9901/listeners
+inbound:127.0.0.1:8000::127.0.0.1:8000
+```
+
+There is a workaround to use `format=json`
+```bash
+$ curl http://localhost:9901/listeners\?format\=json
+{
+ "listener_statuses": [
+  {
+   "name": "inbound:127.0.0.1:8000",
+   "local_address": {
+    "socket_address": {
+     "address": "127.0.0.1",
+     "port_value": 8000
+    }
+   }
+  }
+ ]
+}
+```
+
+#### [Issue #3249](https://github.com/kumahq/kuma/issues/3249): local envoy clusters should have a better name than localhost_<port> 
+
+Today local clusters have name `localhost:<port>` and in stats they're visible as `localhost_<port>`.
+That's not very informative.
+
+#### [Issue #12733](https://github.com/kumahq/kuma/issues/12733): Envoy statPrefix should not use ip addresses
+
+Using IP address in envoy config statPrefix field can cause issues or generate unnecessary changes in metrics.
+This is also problematic for our e2e tests that test envoy config using golden files.
+
+#### [Issue #12093](https://github.com/kumahq/kuma/issues/12093): xds configs, outbound listeners should use the clustername instead of an IP/port combo
+
+We name outbounds like `outbound:10.43.205.116:6379` where IP address doesn't give any useful information.
+
+### Places to use resource identifier
+
+#### URL path
+
+For example `:5681/path-prefix/<identifier>/path-suffix`. 
+Charset is defined by [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3):
+```
+path          = path-abempty    ; begins with "/" or is empty
+              / path-absolute   ; begins with "/" but not "//"
+              / path-noscheme   ; begins with a non-colon segment
+              / path-rootless   ; begins with a segment
+              / path-empty      ; zero characters
+
+path-abempty  = *( "/" segment )
+path-absolute = "/" [ segment-nz *( "/" segment ) ]
+path-noscheme = segment-nz-nc *( "/" segment )
+path-rootless = segment-nz *( "/" segment )
+path-empty    = 0<pchar>
+
+segment       = *pchar
+segment-nz    = 1*pchar
+segment-nz-nc = 1*( unreserved / pct-encoded / sub-delims / "@" )
+              ; non-zero-length segment without any colon ":"
+
+pchar         = unreserved / pct-encoded / sub-delims / ":" / "@"
+
+unreserved    = ALPHA / DIGIT / "-" / "." / "_" / "~"
+pct-encoded   = "%" HEXDIG HEXDIG
+sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
+                 / "*" / "+" / "," / ";" / "="
+```
+
+#### Envoy resource names
+
+For the context, there is already [MADR](036-internal-listeners.md) that regulated the name of internal listeners.
+Also, there was [work](https://docs.google.com/document/d/1OIZK82Tr-4El2FfdlBn7WNRZ7FatkTuEcZKH0FlSTMA/edit?tab=t.0#heading=h.n6cmlf1eel2z) related to Envoy cluster name unification, but it's not finished.
+Discoveries in this work helped me to fill the tables.
+
+There are no restriction on the name format from the Envoy's side.
```

Comment:
Even in length?

Reply:
Envoy doesn't specify a length limit. I tried to create a cluster with the max expected length of resource identifier 
```
253(name) + 253(zone) + 63(mesh) + 63(namespace) + 15(sectionName) + 30(resourcetype) + 3(kri) + 6(_) = 686
```
and it worked as expected.

Reply:
And I think there's no lims for prom label length either: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

Reply:
otel has a length limit defined in the SDK api: https://opentelemetry.io/docs/specs/otel/common/#attribute-limits

However, it states the default in infinity.
We should add these in this MADR for reference.

Reply:
Added in the MADR